client/client-multi/client-c/src/extensions/profile/kaa_profile.c

```diff
@@ -378,7 +382,7 @@ kaa_error_t kaa_profile_handle_server_sync(kaa_profile_manager_t *self
     if (extension_options & KAA_PROFILE_RESYNC_OPTION) {
         self->need_resync = true;
         KAA_LOG_INFO(self->logger, KAA_ERR_NONE, "Going to resync profile...");
-        error_code = kaa_profile_force_sync(self);
+        kaa_profile_try_sync(self);
```

Comment:
Just use void-cast here and there and describe the reason in the comment. An additional function is not needed.